meta-lmp-base/recipes-bsp/efitools/efitools/unlock.patch

```diff
@@ -0,0 +1,190 @@
+ 
++noKEK.esl:
++	> noKEK.esl
++
++ifneq ($(USE_LOCAL_AUTHS),1)
+ noPK.auth: noPK.esl PK.crt sign-efi-sig-list
+ 	$(SIGN_EFI_SIG_LIST) -t "$(shell date --date='1 second' +'%Y-%m-%d %H:%M:%S')" -c PK.crt -k PK.key PK $< $@
+ 
++noKEK.auth: noKEK.esl KEK.crt sign-efi-sig-list
++	$(SIGN_EFI_SIG_LIST) -t "$(shell date --date='1 second' +'%Y-%m-%d %H:%M:%S')" -c KEK.crt -k KEK.key KEK $< $@
++endif
++
+ ms-%.esl: ms-%.crt cert-to-efi-sig-list
+ 	$(CERT_TO_EFI_SIG_LIST) -g $(MSGUID) $< $@
+ 
+@@ -80,6 +93,7 @@ Loader.so: lib/lib-efi.a
+ ReadVars.so: lib/lib-efi.a lib/asn1/libasn1-efi.a
+ UpdateVars.so: lib/lib-efi.a
+ LockDown.so: lib/lib-efi.a
++UnLock.so: lib/lib-efi.a
+ KeyTool.so: lib/lib-efi.a lib/asn1/libasn1-efi.a
+ HashTool.so: lib/lib-efi.a
+ PreLoader.so: lib/lib-efi.a
+@@ -115,7 +129,7 @@ flash-var: flash-var.o lib/lib.a
+ 
+ clean:
+ 	rm -f PK.* KEK.* DB.* DBX.* $(EFIFILES) $(EFISIGNED) $(BINARIES) *.o *.so
+-	rm -f noPK.*
++	rm -f noPK.* noKEK.*
+ 	rm -f doc/*.1
+ 	$(MAKE) -C lib clean
+ 	$(MAKE) -C lib/asn1 clean
+diff --git a/UnLock.c b/UnLock.c
+new file mode 100644
+index 0000000..e67708a
+--- /dev/null
++++ b/UnLock.c
+@@ -0,0 +1,105 @@
++/*
++ * Copyright 2024 <jorge@foundries.io>
++ *
++ * see COPYING file
++ */
++#include <efi.h>
++#include <efilib.h>
++
++#include <variables.h>
++#include <guid.h>
++
++#include "noKEK.h"
++#include "noPK.h"
++
++#define CHECK_EFI_STATUS(efi_status, cert) \
++    if (efi_status != EFI_SUCCESS) \
++        Print(L"Failed to clear %s: %d\n", cert, efi_status); \
++    else \
++        Print(L"%s cleared\n", cert);
++
++#define CHECK_EFI_MODE(efi_status, mode, val) \
++    if (efi_status != EFI_SUCCESS) \
++	Print(L"No %s variable\n", mode); \
++    else \
++	Print(L"Platform is %sin %s\n", val ? L"" :L"not ", mode);
++
++
++EFI_STATUS show_mode()
++{
++	EFI_STATUS efi_status;
++	UINT8 SecureBoot, SetupMode;
++	UINTN DataSize = sizeof(SetupMode);
++
++	efi_status = RT->GetVariable(L"SetupMode", &GV_GUID, NULL, &DataSize,
++				     &SetupMode);
++	CHECK_EFI_MODE(efi_status, L"SetupMode", SetupMode);
++
++
++	efi_status = RT->GetVariable(L"SecureBoot", &GV_GUID, NULL, &DataSize,
++				     &SecureBoot);
++	CHECK_EFI_MODE(efi_status, L"SecureBoot", SecureBoot);
++
++	return EFI_SUCCESS;
++}
++
++EFI_STATUS clear_certificates(void)
++{
++	EFI_STATUS efi_status, ret;
++
++	show_mode();
++
++	efi_status = RT->SetVariable(L"PK", &GV_GUID,
++				     EFI_VARIABLE_NON_VOLATILE
++				     | EFI_VARIABLE_RUNTIME_ACCESS
++				     | EFI_VARIABLE_BOOTSERVICE_ACCESS
++				     | EFI_VARIABLE_TIME_BASED_AUTHENTICATED_WRITE_ACCESS,
++				     noPK_auth_len, noPK_auth);
++	CHECK_EFI_STATUS(efi_status, L"PK");
++	ret = efi_status;
++
++	efi_status = RT->SetVariable(L"KEK", &GV_GUID,
++				     EFI_VARIABLE_NON_VOLATILE
++				     | EFI_VARIABLE_RUNTIME_ACCESS
++				     | EFI_VARIABLE_BOOTSERVICE_ACCESS
++				     | EFI_VARIABLE_TIME_BASED_AUTHENTICATED_WRITE_ACCESS,
++				     noPK_auth_len, noPK_auth);
```

Comment:
Shouldn't this be noKEK_auth instead?

Reply:
no, not on this case since the payload is empty and PK and KEK are both signed with PK.